fw/t/unit/test_http_parser_common.h

```diff
@@ -0,0 +1,710 @@
+#include "fuzzer.h"
+
+#ifdef EXPORT_SYMBOL
+#undef EXPORT_SYMBOL
+#endif
+#define EXPORT_SYMBOL(...)
+
+#include "msg.h"
+#include "str.h"
+#include "http_msg.h"
+#include "http_parser.h"
+#include "http_sess.h"
+
+static const unsigned int CHUNK_SIZES[] = { 1, 2, 3, 4, 8, 16, 32, 64, 128,
+                                   256, 1500, 9216, 1024*1024
+                                  /* to fit a message of 'any' size */
+                                 };
+static unsigned int chunk_size_index = 0;
+#define CHUNK_SIZE_CNT ARRAY_SIZE(CHUNK_SIZES)
+
+enum {
+	CHUNK_OFF,
+	CHUNK_ON
+};
+
+#define TOKEN_ALPHABET		"!#$%&'*+-.0123456789ABCDEFGHIJKLMNOPQ"	\
+				"RSTUVWXYZ^_`abcdefghijklmnopqrstuvwxyz|~"
+#define QETOKEN_ALPHABET	TOKEN_ALPHABET "\"="
+#define OTHER_DELIMETERS	"(),/:;<=>?@[\\]{}"
+#define OBS_TEXT		"\x80\x90\xC8\xAE\xFE\xFF"
+#define ETAG_ALPHABET		OTHER_DELIMETERS TOKEN_ALPHABET OBS_TEXT
+#define VCHAR_ALPHABET		"\x09 \"" OTHER_DELIMETERS 		\
+				TOKEN_ALPHABET OBS_TEXT
+
+#define EXPECT_BLOCK_DIGITS(head, tail, BLOCK_MACRO)		\
+	BLOCK_MACRO(head tail);					\
+	BLOCK_MACRO(head "  " tail);				\
+	BLOCK_MACRO(head "5a" tail);				\
+	BLOCK_MACRO(head "\"" tail);				\
+	BLOCK_MACRO(head "=" tail);				\
+	BLOCK_MACRO(head "-1" tail);				\
+	BLOCK_MACRO(head "0.99" tail);				\
+	BLOCK_MACRO(head "dummy" tail);				\
+	BLOCK_MACRO(head "4294967296" tail);			\
+	BLOCK_MACRO(head "9223372036854775807" tail);		\
+	BLOCK_MACRO(head "9223372036854775808" tail);		\
+	BLOCK_MACRO(head "18446744073709551615" tail);		\
+	BLOCK_MACRO(head "18446744073709551616" tail)
+
+#define EXPECT_BLOCK_SHORT(head, tail, BLOCK_MACRO)		\
+	BLOCK_MACRO(head "65536" tail);				\
+	BLOCK_MACRO(head "2147483647" tail);			\
+	BLOCK_MACRO(head "2147483648" tail);			\
+	BLOCK_MACRO(head "4294967295" tail)
+
+/* For ETag and If-None-Match headers */
+#define COMMON_ETAG_BLOCK(head, BLOCK_MACRO)			\
+	BLOCK_MACRO(head "\"dummy");				\
+	BLOCK_MACRO(head "dummy\"");				\
+	BLOCK_MACRO(head "'dummy'");				\
+	BLOCK_MACRO(head "W/ \"dummy\"");			\
+	BLOCK_MACRO(head "w/\"dummy\"");			\
+	BLOCK_MACRO(head "\"\x00\"");				\
+	BLOCK_MACRO(head "\"\x0F\"");				\
+	BLOCK_MACRO(head "\"\x7F\"");				\
+	BLOCK_MACRO(head "\" \"");				\
+	BLOCK_MACRO(head "\"\"\"")
+
+static TfwHttpReq *req, *sample_req;
+static TfwHttpResp *resp;
+static TfwH2Conn conn;
+static TfwStream stream;
+static size_t hm_exp_len = 0;
+
+typedef struct data_rec
+{
+	char *buf;
+	size_t size;
+} TfwDataRec;
+
+typedef struct header_rec
+{
+	TfwDataRec name;
+	TfwDataRec value;
+} TfwHeaderRec;
+
+typedef struct frame_rec
+{
+    unsigned int len;
+    unsigned char *str;
+    TfwFrameType subtype;	// used only for FUZZ_REQ_H2 cases
+} TfwFrameRec;
+
+#define ALLOWED_FRAMES_CNT 2
+
+static TfwFrameRec frames[ALLOWED_FRAMES_CNT];
+static unsigned int frames_cnt = 0;
+static unsigned int frames_max_sz = 0;
+static unsigned int frames_total_sz = 0;
+static TfwHeaderRec tmp_header_rec __attribute__((unused));
```

Comment:
I will remove that struct in [#1612](https://github.com/tempesta-tech/tempesta/pull/1612)